lib/web_ui/lib/src/engine/embedder.dart

```diff
@@ -180,102 +133,32 @@ class FlutterViewEmbedder {
       '$defaultFontStyle $defaultFontWeight ${defaultFontSize}px $defaultFontFamily';
 
   void reset() {
-    final bool isWebKit = browserEngine == BrowserEngine.webkit;
-
-    _styleElement?.remove();
-    _styleElement = createDomHTMLStyleElement();
-    _resourcesHost?.remove();
-    _resourcesHost = null;
-    domDocument.head!.append(_styleElement!);
-    final DomCSSStyleSheet sheet = _styleElement!.sheet! as DomCSSStyleSheet;
-    applyGlobalCssRulesToSheet(
-      sheet,
-      browserEngine: browserEngine,
-      hasAutofillOverlay: browserHasAutofillOverlay(),
-    );
-
-    final DomHTMLBodyElement bodyElement = domDocument.body!;
-
-    bodyElement.setAttribute(
+    _applicationDom.setHostAttribute(
       'flt-renderer',
       '${renderer.rendererTag} (${FlutterConfiguration.flutterWebAutoDetect ? 'auto-selected' : 'requested explicitly'})',
     );
-    bodyElement.setAttribute('flt-build-mode', buildMode);
-
-    setElementStyle(bodyElement, 'position', 'fixed');
-    setElementStyle(bodyElement, 'top', '0');
-    setElementStyle(bodyElement, 'right', '0');
-    setElementStyle(bodyElement, 'bottom', '0');
-    setElementStyle(bodyElement, 'left', '0');
-    setElementStyle(bodyElement, 'overflow', 'hidden');
-    setElementStyle(bodyElement, 'padding', '0');
-    setElementStyle(bodyElement, 'margin', '0');
-
-    // TODO(yjbanov): fix this when KVM I/O support is added. Currently scroll
-    //                using drag, and text selection interferes.
-    setElementStyle(bodyElement, 'user-select', 'none');
-    setElementStyle(bodyElement, '-webkit-user-select', 'none');
-    setElementStyle(bodyElement, '-ms-user-select', 'none');
-    setElementStyle(bodyElement, '-moz-user-select', 'none');
-
-    // This is required to prevent the browser from doing any native touch
-    // handling. If this is not done, the browser doesn't report 'pointermove'
-    // events properly.
-    setElementStyle(bodyElement, 'touch-action', 'none');
-
-    // These are intentionally outrageous font parameters to make sure that the
-    // apps fully specify their text styles.
-    setElementStyle(bodyElement, 'font', defaultCssFont);
-    setElementStyle(bodyElement, 'color', 'red');
-
+    _applicationDom.setHostAttribute('flt-build-mode', buildMode);
+    _applicationDom.setHostAttribute('flt-application-dom', _applicationDom.type);
```

Comment:
I think each implementation can set this attribute privately. I'm not sure there's value in having them be set from outside. As a bonus, it will also obviate the need for the `type` getter.

Reply:
> I'm not sure there's value in having them be set from outside.

I was trying to rely less in global variables (like the buildMode or the type of renderer), and more in incoming parameters.

The ViewEmbedder already uses these globals, but there's no need for those globals to spread to more places than strictly needed :P

Reply:
Removed this, I'm still passing values in from the embedder, but as a "metadata" map.